src/core/Form/ErrorSummary/ErrorSummary.tsx

```diff
@@ -0,0 +1,178 @@
+} from '../../theme/utils/spacing';
+import { baseStyles } from './ErrorSummary.baseStyles';
+import { Heading } from '../../Heading/Heading';
+import { Link } from '../../Link';
+
+const baseClassName = 'fi-error-summary';
+const inlineAlertClassNames = {
+  styleWrapper: `${baseClassName}_style-wrapper`,
+  content: `${baseClassName}_content`,
+  heading: `${baseClassName}_heading`,
+  textContentWrapper: `${baseClassName}_text-content-wrapper`,
+  icon: `${baseClassName}_icon`,
+  smallScreen: `${baseClassName}--small-screen`,
+};
+
+interface BasicErrorSummaryItemProps {
+  /** Visible text of the error item */
+  text: string;
+}
+
+type InputReferenceProps =
+  | {
+      /** HTML id of an input element. The error corresponds to this input. Provide either this or `inputRef`. */
+      inputId: string;
+      /** Ref pointing to an input element. The error corresponds to this input. Provide either this or `inputId` */
+      inputRef?: never;
+    }
+  | {
+      inputId?: never;
+      inputRef: React.RefObject<HTMLElement>;
+    };
+
+export type ErrorSummaryItemProps = BasicErrorSummaryItemProps &
+  InputReferenceProps;
+
+export interface ErrorSummaryProps extends HtmlDivProps, MarginProps {
+  /**
+   * List of error messages.
+   * Each item is rendered as a link and represents an error message concerning one input field.
+   * When user clicks a link, the input associated with it is focused.
+   */
+  items?: Array<ErrorSummaryItemProps>;
+  /** Toggles small screen styling */
+  smallScreen?: boolean;
+  /** Heading for the error summary */
+  headingText: ReactNode;
+  /** Semantic level of the heading. Does not affect visual styling.
+   * @default 'h3'
+   */
+  headingVariant?: Exclude<hLevels, 'h1'>;
+  /** Ref for the summary's heading. Used in applying focus to the heading when the form is submitted. */
+  headingRef: React.Ref<HTMLHeadingElement>;
+  /** Ref is placed to the outermost div element of the component. Alternative for React `ref` attribute. */
+  forwardedRef?: React.Ref<HTMLDivElement>;
+}
+
+const BaseErrorSummary = (props: ErrorSummaryProps) => {
+  const {
+    className,
+    headingText,
+    headingVariant = 'h3',
+    headingRef,
+    items,
+    smallScreen,
+    id,
+    forwardedRef,
+    ...rest
+  } = props;
+  const [marginProps, passProps] = separateMarginProps(rest);
+  const marginStyle = spacingStyles(marginProps);
+
+  const focusInput = (
+    event: MouseEvent<HTMLAnchorElement>,
+    errorItem: ErrorSummaryItemProps,
+  ) => {
+    // Prevents the normal browser behavior when clicking on an anchor
+    event.preventDefault();
+
+    if (errorItem.inputRef && errorItem.inputRef.current) {
+      errorItem.inputRef.current.focus();
+    } else if (errorItem.inputId && errorItem.inputId !== '') {
+      const element = document.getElementById(errorItem.inputId);
+      if (element) {
+        element.focus();
+      }
+    }
+  };
+
+  return (
+    <HtmlDivWithRef
+      as="section"
```

Comment:
What does the `as="section"` accomplish here? It seems to make it a landmark <section> element in the DOM, but there's no `aria-label` or `aria-labelledby` attached to it.

Reply:
Well, technically the aria-props from `passProps` go here, which gives it some use.

Reply:
Let's leave it as is for the moment and maybe see if linking the heading to the section could be beneficial in the future.